app/src/common/shared/com/igalia/wolvic/browser/engine/Session.java

```diff
@@ -1484,149 +1434,115 @@ public void onContentLoaded(@NonNull GeckoSession geckoSession, @NonNull Content
+    public WResult<PromptResponse> onBeforeUnloadPrompt(@NonNull WSession aSession, @NonNull BeforeUnloadPrompt prompt) {
         if (mPromptDelegate != null) {
             return mPromptDelegate.onBeforeUnloadPrompt(aSession, prompt);
         }
-        return GeckoResult.fromValue(prompt.dismiss());
+        return WResult.fromValue(prompt.dismiss());
     }
 
     @Nullable
     @Override
-    public GeckoResult<PromptResponse> onLoginSelect(@NonNull GeckoSession geckoSession, @NonNull AutocompleteRequest<Autocomplete.LoginSelectOption> autocompleteRequest) {
+    public WResult<PromptResponse> onLoginSelect(@NonNull WSession aSession, @NonNull AutocompleteRequest<WAutocomplete.LoginSelectOption> autocompleteRequest) {
         if (mPromptDelegate != null) {
-            return mPromptDelegate.onLoginSelect(geckoSession, autocompleteRequest);
+            return mPromptDelegate.onLoginSelect(aSession, autocompleteRequest);
         }
-        return GeckoResult.fromValue(autocompleteRequest.dismiss());
+        return WResult.fromValue(autocompleteRequest.dismiss());
     }
 
     @Nullable
     @Override
-    public GeckoResult<PromptResponse> onLoginSave(@NonNull GeckoSession geckoSession, @NonNull AutocompleteRequest<Autocomplete.LoginSaveOption> autocompleteRequest) {
+    public WResult<PromptResponse> onLoginSave(@NonNull WSession aSession, @NonNull AutocompleteRequest<WAutocomplete.LoginSaveOption> autocompleteRequest) {
         if (mPromptDelegate != null) {
-            return mPromptDelegate.onLoginSave(geckoSession, autocompleteRequest);
-        }
-        return GeckoResult.fromValue(autocompleteRequest.dismiss());
-    }
-
-    // MediaDelegate
-
-    /*@Override
-    public void onMediaAdd(@NonNull GeckoSession aSession, @NonNull MediaElement element) {
-        if (mState.mSession != aSession) {
-            return;
-        }
-        Media media = new Media(element);
-        mState.mMediaElements.add(media);
-
-        for (VideoAvailabilityListener listener: mVideoAvailabilityListeners) {
-            listener.onVideoAvailabilityChanged(media, true);
+            return mPromptDelegate.onLoginSave(aSession, autocompleteRequest);
         }
+        return WResult.fromValue(autocompleteRequest.dismiss());
     }
 
-    @Override
-    public void onMediaRemove(@NonNull GeckoSession aSession, @NonNull MediaElement element) {
-        if (mState.mSession != aSession) {
-            return;
-        }
-        for (int i = 0; i < mState.mMediaElements.size(); ++i) {
-            Media media = mState.mMediaElements.get(i);
-            if (media.getMediaElement() == element) {
-                media.unload();
-                mState.mMediaElements.remove(i);
-                for (VideoAvailabilityListener listener: mVideoAvailabilityListeners) {
-                    listener.onVideoAvailabilityChanged(media, false);
-                }
-                return;
-            }
-        }
-    }*/
```

Comment:
Where are these two removed now?

Reply:
It's a leftover from an old API. Before using MesiaSession Gecko exposed an API called MediaElement